airflow/sensors/sql_sensor.py

```diff
@@ -61,14 +58,13 @@ class SqlSensor(BaseSensorOperator):
 
     @apply_defaults
     def __init__(self, conn_id, sql, parameters=None, success=None, failure=None, fail_on_empty=False,
-                 allow_null=True, *args, **kwargs):
```

Comment:
I would say leave this as an option but set just change the default please.

Reply:
@ashb I strongly disagree. Why would we leave this parameter? First of all, I think that majority of users would never use this option if NULL is 'keep poking' result for sensor. For those who want to leave this weird sensor behavior there is still an option via 'success' parameter -- it is described in changelist.
Long story short: I don't think we should support this bug in new version where we can break compatibility.